test/python/test_tools_tiger_data.py

```diff
@@ -0,0 +1,45 @@
+"""
+Test for tiger data function
+"""
+from pathlib import Path
+
+import pytest
+import tarfile
+
+from nominatim.tools import tiger_data, database_import
+
+
+@pytest.mark.parametrize("threads", (1, 5))
+def test_add_tiger_data(dsn, src_dir, def_config, monkeypatch,tmp_path,
+                        temp_db_cursor, threads, temp_db):
+    monkeypatch.setenv('NOMINATIM_DATABASE_MODULE_PATH', '.')
+    temp_db_cursor.execute('CREATE EXTENSION hstore')
+    temp_db_cursor.execute('CREATE EXTENSION postgis')
+    temp_db_cursor.execute('CREATE TABLE place (id INT)')
+
+    database_import.import_base_data('dbname=' + temp_db, src_dir / 'data',
+                                     ignore_partitions=False)
```

Comment:
There is a [fixture sql_preprocessor](https://github.com/osm-search/Nominatim/blob/35f4695b67b50a41caf69378ea13044ea91f1d2c/test/python/test_db_sql_preprocessor.py#L26), which does the minimum setup required to be able to use the SQLPreprocessor class. If you move that into `conftest.py`, you can use the fixture here as well. That would replace the setenv and import_base_data. Less typing and when the requirements for the SQLPreprocessor change later, there is only one function that needs adapting.